Sources/StreamChat/Repositories/AuthenticationRepository.swift

```diff
@@ -140,6 +153,98 @@ class AuthenticationRepository {
         scheduleTokenFetch(userInfo: nil, tokenProvider: tokenProviderCheckingSuccess, completion: completion)
     }
 
+    func prepareEnvironment(
+        userInfo: UserInfo?,
+        newToken: Token,
+        completion: @escaping (Error?) -> Void
+    ) {
+        let state: EnvironmentState
+        if currentUserId == nil {
+            state = .firstConnection
+        } else if newToken.userId == currentUserId {
+            state = .newToken
+        } else {
+            state = .newUser
+        }
+
+        log.assert(delegate != nil, "Delegate should not be nil at this point")
+
+        switch state {
+        case .firstConnection:
```

Comment:
The handlings for both `firstConnection` and `newToken` are the same. Maybe we can put them in the same case (or extract these lines in a method)?

Reply:
Oh wow, did not realize 😂 After abstracting the logic, there are no more differences